besu/src/main/java/org/hyperledger/besu/cli/subcommands/blocks/BlocksSubCommand.java

```diff
@@ -150,37 +159,48 @@ public void run() {
     @Override
     public void run() {
       parentCommand.parentCommand.configureLogging(false);
-      LOG.info("Import {} block data from {}", format, blocksImportFile);
 
       checkCommand(parentCommand);
       checkNotNull(parentCommand.rlpBlockImporter);
       checkNotNull(parentCommand.jsonBlockImporterFactory);
-
+      if (blockImportFileOption.isEmpty()) {
+        throw new ParameterException(spec.commandLine(), "No files specified to import.");
+      }
+      LOG.info("Import {} block data from {} files", format, blockImportFiles.size());
       final Optional<MetricsService> metricsService = initMetrics(parentCommand);
 
-      try {
-        // As blocksImportFile even if initialized as null is injected by PicoCLI and param is
-        // mandatory. So we are sure it's always not null, we can remove the warning.
-        //noinspection ConstantConditions
-        final Path path = blocksImportFile.toPath();
-        final BesuController<?> controller = createController();
-        switch (format) {
-          case RLP:
-            importRlpBlocks(controller, path);
-            break;
-          case JSON:
-            importJsonBlocks(controller, path);
-            break;
-          default:
-            throw new ParameterException(
-                spec.commandLine(), "Unsupported format: " + format.toString());
+      try (final BesuController<?> controller = createController()) {
+        for (final Path path : blockImportFiles) {
+          try {
+            LOG.info("Importing from {}", path);
+
+            switch (format) {
+              case RLP:
+                importRlpBlocks(controller, path);
+                break;
+              case JSON:
+                importJsonBlocks(controller, path);
+                break;
+              default:
+                throw new ParameterException(
+                    spec.commandLine(), "Unsupported format: " + format.toString());
```

Comment:
Would it be worth it to indicate that this is a developer exception and ask for a bug report? I think this can only happen if there's an enumerated block format that we didn't make a case clause for here.

Reply:
Despite the green text I didn't write this block, it's the beneficiary of being indented in a for loop.

But if the concern is developer mistake for new unenumerated formats the better approach IMHO is to remove the default block and have the build break at compile time and make the author consciously think about what they want to happen.  Being an enum Java will let us not have a default if we fully enumerate the enum.

Reply:
Sounds good to me